lib/models/application.coffee

```diff
@@ -694,6 +694,180 @@ getApplicationModel = (deps, opts) ->
+	# @name isTrackingLatestRelease
+	# @public
+	# @function
+	# @memberof resin.models.application
+	#
+	# @param {String|Number} nameOrId - application name (string) or id (number)
+	# @fulfil {Boolean} - is tracking the latest release
+	# @returns {Promise}
+	#
+	# @example
+	# resin.models.application.isTrackingLatestRelease('MyApp').then(function(isEnabled) {
+	# 	console.log(isEnabled);
+	# });
+	#
+	# @example
+	# resin.models.application.isTrackingLatestRelease(123).then(function(isEnabled) {
+	# 	console.log(isEnabled);
+	# });
+	#
+	# @example
+	# resin.models.application.isTrackingLatestRelease('MyApp', function(error, isEnabled) {
+	# 	console.log(isEnabled);
+	# });
+	###
+	exports.isTrackingLatestRelease = (nameOrId, callback) ->
+		exports.get(nameOrId, $select: 'should_track_latest_release')
+		.get('should_track_latest_release')
+		.asCallback(callback)
+
+	###*
+	# @summary Set a specific application to run a particular release
+	# @name setToRelease
+	# @public
+	# @function
+	# @memberof resin.models.application
+	#
+	# @description Configures the application to run a particular release
+	# and not get updated when the latest release changes.
+	#
+	# @param {String|Number} nameOrId - application name (string) or id (number)
+	# @param {String} fullReleaseHash - the hash of a successful release (string) or id (number)
+	# @returns {Promise}
+	#
+	# @example
+	# resin.models.application.setToRelease('MyApp', 'f7caf4ff80114deeaefb7ab4447ad9c661c50847').then(function() {
+	# 	...
+	# });
+	#
+	# @example
+	# resin.models.application.setToRelease(123, 'f7caf4ff80114deeaefb7ab4447ad9c661c50847').then(function() {
+	# 	...
+	# });
+	#
+	# @example
+	# resin.models.application.setToRelease('MyApp', 'f7caf4ff80114deeaefb7ab4447ad9c661c50847', function(error) {
+	# 	if (error) throw error;
+	# 	...
+	# });
+	###
+	exports.setToRelease = (nameOrId, fullReleaseHash, callback) ->
+		getId(nameOrId)
+		.then (applicationId) ->
+			pine.patch
+				resource: 'application'
+				id: applicationId
+				body:
+					commit: fullReleaseHash
+					should_track_latest_release: false
+		.asCallback(callback)
+
+	###*
+	# @summary Get the hash of the current release for a specific application
+	# @name getTrackedReleaseHash
+	# @public
+	# @function
+	# @memberof resin.models.application
+	#
+	# @param {String|Number} nameOrId - application name (string) or id (number)
+	# @fulfil {String} - The release hash of the current release
+	# @returns {Promise}
+	#
+	# @example
+	# resin.models.application.getTrackedReleaseHash('MyApp').then(function(release) {
+	# 	console.log(release);
+	# });
+	#
+	# @example
+	# resin.models.application.getTrackedReleaseHash(123).then(function(release) {
+	# 	console.log(release);
+	# });
+	#
+	# @example
+	# resin.models.application.getTrackedReleaseHash('MyApp', function(release) {
+	# 	console.log(release);
+	# });
+	###
+	exports.getTrackedReleaseHash = (nameOrId, callback) ->
```

Comment:
I think `tracked` is a little confusing here. We use it to talk about tracking the latest release elsewhere, to describe following one thing (the latest), that changes what it points to. It's a small thing, but it does feel a bit unclear. What we want to know here is not what we're following, but what the current target of that is.

How about `getCurrentReleaseHash`? That matches the current dashboard designs too (which put `Current` next to the current release).

Alternatively imo `Target` would work well, but we should use that consistently everywhere, and we don't use it right now.

Reply:
I agree that we can do better than `tracked`. The new designs only use it as `track: latest`.
We will probably keep it for later when we add release channels.
The only thing I don't like about `current` is that it isn't accurate in the context of a device, were we should better use the same method name for consistency. (A device can be made to truck release B but is still running release A. In that case `getCurrentReleaseHash` probably means A but the method we implement here is meant to return B, so Tracked/Target make more sense.)